staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -351,7 +359,20 @@ func (s preparedGenericAPIServer) Run(stopCh <-chan struct{}) error {
 	}()
 
 	// close socket after delayed stopCh
-	stoppedCh, listenerStoppedCh, err := s.NonBlockingRun(delayedStopCh.Signaled())
+	drainedCh := s.terminationSignals.InFlightRequestsDrained
+	stopHttpServerCh := delayedStopCh.Signaled()
+	shutdownTimeout := s.ShutdownTimeout
+	if s.KeepListeningDuringGracefulTermination {
+		// when this mode is enabled, we do the following:
+		// - the server will continue to listen until all existing
+		//   requests in flight have been drained
+		// - once drained, http Server Shutdown is invoked with a timeout of 5s
+		stopHttpServerCh = drainedCh.Signaled()
+		shutdownTimeout = 5 * time.Second
```

Comment:
it changes the previous behavior, it is not backward compatible, not sure it has to be
- it changes the maximum time from `60s` to `65s`
- doesn't apply the timeout set in `ShutdownTimeout` (could be set to 10s)